lib/lowkiq.rb

```diff
@@ -64,9 +64,13 @@ def server_wrapper
       end
     end
 
+    def client_wrapper
+      @client_middleware ||= self.middleware_wrapper(self.client_middlewares)
+    end
+
     def shard_handlers
       self.workers.flat_map do |w|
-        ShardHandler.build_many w, self.server_wrapper
+        ShardHandler.build_many w, self.middleware_wrapper(self.server_middlewares)
```

Comment:
I think you should keep `self.server_wrapper` and implement it like `client_wrapper`.

Reply:
Refactored the code as suggested.